backup_github.py

```diff
@@ -31,114 +40,148 @@ def __enter__(self):
         if os.path.exists(repo_path):
-            logging.info("The repository {} already exists on destination. Pulling " \
-                     "all branches".format(repo.name))
+            logger.info("The repository {} already exists on destination. "
+                        "Pulling all branches".format(repo.name))
             with cd(repo_path):
                 try:
-                    #These stdout and stderr flush out the normal github output
-                    #the alternative of using -q doesn't always work
+                    # These stdout and stderr flush out the normal github
+                    # output the alternative of using -q doesn't always work
                     check_call(['git', 'stash'], stdout=PIPE, stderr=STDOUT)
                     check_call(['git', 'pull'], stdout=PIPE, stderr=STDOUT)
-		#General exception to better catch errors
+                # General exception to better catch errors
                 except CalledProcessError:
-		    print "ERROR: There was an error fetching the branches from " \
-                              "the repository {}, skipping it".format(repo.name)
-                    logging.error("There was an error fetching the branches from " \
-                              "the repository {}, skipping it".format(repo.name))
+                    logger.error("There was an error fetching the branches "
+                                 "from the repository {}, "
+                                 "skipping it".format(repo.name))
                     pass
-	    logging.info("Finished copying repo {}".format(repo.name))
-	    print "Finished copying repo {}".format(repo.name)
-        #Otherwise clone the repository and fetch all branches
+            logger.info("Finished copying repo {}".format(repo.name))
+        # Otherwise clone the repository and fetch all branches
         else:
-            logging.info("The repository {} isn't cloned at {}, cloning instead of updating...".format(repo.name, repo_path))
+            logger.info("The repository {} isn't cloned at {}, cloning instead"
+                        " of updating...".format(repo.name, repo_path))
             try:
-                check_call(['git', 'clone', source, repo_path], stdout=PIPE, stderr=STDOUT)
-                logging.info("Cloning {}".format(repo.name))
-            except CalledProcessError:
-                print 'ERROR: Problem cloning repository {}, skipping it'.format(repo.name)
-                logging.error('ERROR: Error cloning repository {}, skipping it'.format(repo.name))
+                check_call(['git', 'clone', source, repo_path],
+                           stdout=PIPE, stderr=STDOUT)
+                logger.info("Cloning {}".format(repo.name))
+            except CalledProcessError as e:
+                logger.error("ERROR: Error cloning repository {}, "
+                             "skipping it".format(repo.name))
+                logger.error(str(e))
             pass
```

Comment:
Should this be indented one more time? 🤔

Reply:
I think it can be deleted - doesn't look like it is doing anything to me..

Reply:
Very good catch!

Reply:
@ewels, true. `except-pass` doesn't do anything else than the default for just `except` right?

Reply:
It’s not even that. `pass` does literally nothing, but you need something after `except` to avoid an indentation error. So if you have any code there at all it’s not necessary (eg. a log message)

Reply:
> It’s not even that

I mean, it’s not a default (assuming that a default usually does something)